fedimint-bin-tests/src/main.rs

```diff
@@ -563,21 +565,368 @@ async fn cli_tests(dev_fed: DevFed) -> Result<()> {
         "{data_dir}/server-0/config-plaintext-2.json",
     );
 
-    fed.pegin(10_000).await?;
-    // fed.pegin_gateway(99_999).await?;
+    fed.pegin(10_000).await;
+    fed.pegin_gateway(99_999, &gw_cln).await;
 
-    let connect_string = fs::read_to_string(format!("{data_dir}/client-connect")).await?;
-    fs::remove_file(format!("{data_dir}/client.json"));
+    let connect_string = fs::read_to_string(format!("{}/client-connect", data_dir)).await?;
+    fs::remove_file(format!("{}/client.json", data_dir));
     cmd!(fed, "join-federation", connect_string.clone());
 
     let fed_id = fed.federation_id().await;
     assert_eq!(
-        cmd!(fed, "decode-connect-info", connect_string)
+        cmd!(fed, "decode-connect-info", connect_string.clone())
             .out_json()
             .await?["id"]
             .as_str(),
         Some(&*fed_id),
     );
+    // Number required for one honest is ceil(($FM_FED_SIZE-1)/3+1)
+    let one_honest = 2;
```

Comment:
`one_honest_threashold`. I never like that "one honest" as integer thing. It's a bit `one = 2`, to me.

Reply:
@jkitman You're probably the best person to talk about it. You see my point (here and below)?

Reply:
I'm changing the connect info to be one per guardian in order to accommodate the join tokens.  Would like to get the changes in https://github.com/fedimint/fedimint/pull/2175 merged before this PR.

Reply:
@GetPsyched Sorry for breaking these tests, if you look at the `cli-test.sh` in master you will see how to fix it.

Reply:
No worries, I've added your changes and everything should work fine.